rpc/query_server.go

```diff
@@ -327,7 +328,7 @@ func (s *QueryServer) EthGetCode(address eth.Data, block eth.BlockHeight) (eth.D
 	snapshot := s.StateProvider.ReadOnlyState()
 	defer snapshot.Release()
 
-	evm := levm.NewLoomVm(snapshot, nil, nil, nil, false)
+	evm := levm.NewLoomVm(snapshot, s.EvmStore, nil, nil, nil, false)
```

Comment:
This doesn't look right to me, the EVM created to handle this query will access the same `trie.Database` as the one being written to in the tx/middleware handlers, the `EvmStore.trieDB` isn't a snapshot. The `trie.Database` looks thread-safe, but that may not be sufficient to guarantee correctness. Also the `EvmStore.trieDB` should probably be accessed via atom ops like the `MultiWriterAppStore.lastSavedTree`.